### DISCUSSION TITLE
with table partition key, how to use field value prefix

### DISCUSSION THREAD

**Discussion by flyly0755** [ClickHouse/ClickHouse#42753] [Q&A] [answered]
There is one table, i want to use one field value prefix as partiton key, for example, the field called concatname,
the value maybe abc_xyz, def_uvw, and so on
i want to use the value prefix(the first 3 chars such as abc, def) as partition key.
use which function to get the prefix?
some like this:
CREATE TABLE test_tb
(
    `concatname` String,
		...
)
ENGINE = MergeTree
PARTITION BY concatname[0:3] --- the right syntax is ?

**Comment by mo-avatar** [comment, marked as answer]
substring or splitByChar may help you with your problem, please search how to use it in documentation.

**Reply by flyly0755** [reply]
Seems  there is problem with substring, when insert into data to the table with substring function as partition,
raise error:
Code: 135. DB::Exception: Indices in strings are 1-based: while executing 'FUNCTION substring(xxx: 0, 0 :: 1, 3 :: 2) -> substring(xxx, 0, 3) String : 3'. (ZERO_ARRAY_OR_TUPLE_INDEX) (version 21.12.3.32 (official build))

**Reply by mo-avatar** [reply]
the sql standard of substring function start the indexing from one not zero. In other words, you should write it like  substring(x, 1, 3)
